es/es-repository-hdfs/build.gradle

```diff
@@ -0,0 +1,111 @@
+    compile "org.apache.hadoop:hadoop-annotations:${versions.hadoop2}"
+    compile "org.apache.hadoop:hadoop-auth:${versions.hadoop2}"
+    compile "org.apache.hadoop:hadoop-hdfs:${versions.hadoop2}"
+    compile 'org.apache.htrace:htrace-core:3.1.0-incubating'
+    compile "com.google.guava:guava:${versions.guava}"
+    compile 'com.google.protobuf:protobuf-java:2.5.0'
+    compile 'commons-logging:commons-logging:1.1.3'
+    compile 'commons-cli:commons-cli:1.2'
+    compile 'commons-collections:commons-collections:3.2.2'
+    compile ('commons-configuration:commons-configuration:1.6') {
+        exclude group: 'commons-beanutils', module: 'commons-beanutils-core'
+    }
+    compile 'commons-io:commons-io:2.4'
+    compile 'commons-lang:commons-lang:2.6'
+    compile 'javax.servlet:servlet-api:2.5'
+    compile "org.slf4j:slf4j-api:${versions.slf4j}"
+
+    extraLibs "org.apache.hadoop:hadoop-client:${versions.hadoop2}"
+    extraLibs "org.apache.hadoop:hadoop-common:${versions.hadoop2}"
+    extraLibs "org.apache.hadoop:hadoop-annotations:${versions.hadoop2}"
+    extraLibs "org.apache.hadoop:hadoop-auth:${versions.hadoop2}"
+    extraLibs "org.apache.hadoop:hadoop-hdfs:${versions.hadoop2}"
+    extraLibs 'org.apache.htrace:htrace-core:3.1.0-incubating'
+    extraLibs 'com.google.protobuf:protobuf-java:2.5.0'
+    extraLibs 'commons-collections:commons-collections:3.2.2'
+    extraLibs ('commons-configuration:commons-configuration:1.6') {
+        exclude group: 'commons-beanutils', module: 'commons-beanutils-core'
+    }
+    extraLibs 'commons-io:commons-io:2.4'
+    extraLibs 'commons-lang:commons-lang:2.6'
+    extraLibs 'javax.servlet:servlet-api:2.5'
+}
+
+def pluginDir = file('../upstream/plugins/repository-hdfs')
+
+sourceSets {
+    main {
+        java {
+            srcDir "$pluginDir/src/main/java"
+        }
+        output.resourcesDir = null
+    }
+    test {
+        output.resourcesDir = null
+        output.classesDir = null
+    }
+}
+
+idea {
+    module {
+        contentRoot = pluginDir
+        excludeDirs += files('build', 'src/test')
+    }
+}
+
+jar {dependsOn(['writePropertiesFileRepositoryHSFS', 'copyToLib'])}
+
+task writePropertiesFileRepositoryHSFS << {
+    def file = new File(buildDir.path + '/plugin-descriptor.properties')
+    def parentDir = new File(file.parent)
+    parentDir.mkdirs()
+    file.createNewFile()
+
+    ant.propertyfile(file: file.path, comment: 'Plugin Descriptor') {
+        ant.entry(key: 'name', value: project.name)
+        ant.entry(key: 'description', value: project.description)
+        ant.entry(key: 'version', value: plugin_repositoryhdfs_version)
+        ant.entry(key: 'site', value: true)
+        ant.entry(key: 'elasticsearch.version', value: "${versions.elasticsearch}")
+        ant.entry(key: 'java.version', value: project.targetCompatibility)
+        ant.entry(key: 'classname', value: classname)
+    }
+}
+
+task copyToLib(type: Copy) {
+    into buildDir.path + '/lib'
```

Comment:
we dont want to ship crate with all hadoop libs. if the user wants to use it, it have to copy the libs by himself as the current documentation also states